### PR TITLE
Switch from acorn to meriyah

### LINKED ISSUE #41: Switch from acorn to meriyah
Switch from acorn to meriyah.

- [x] Convert to meriyah parsing and tree. https://github.com/knockout-lint/knockout-lint/blob/7504cf32fb3ced69c592fb34fc9c5de6733f3c96/src/parser/compile-bindings.ts#L41-L74
- [x] Change acorn in the "built with" section in the readme file to meriyah (May be outdated when checked out)

### PR DESCRIPTION
**Pull request by tscpp**
Switch from parsing with Acorn to Meriyah. Meriyah is about 300% faster than acorn and support ES2020 as of 2020/11.

Fixes #41

### CODE REVIEW

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
+   // Use IntelliSense to learn about possible attributes.
+   // Hover to view descriptions of existing attributes.
+   // For more information, visit: https://go.microsoft.com/fwlink/?linkid=830387
```
Comment (tscpp) [inline .vscode/launch.json:4]:
```suggestion
```

**Comment by tscpp** [comment]
https://github.com/meriyah/meriyah/issues/156

File: .vscode/launch.json [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+         "skipFiles": [
+            "<node_internals>/**"
+         ],
+         "program": "${workspaceFolder}/demo/build/lint.test.js",
+         "outFiles": [
+            "${workspaceFolder}/**/*.js"
+         ]
+      }
+   ]
+}
```
Comment (maskmaster) [inline .vscode/launch.json:17]: Intentionally in this PR?

File: package.json [hunk 3]
```diff
@@ -19,6 +19,7 @@
 	"dependencies": {
 		"glob": "^7.1.6",
 		"knockout": "^3.5.1",
+		"meriyah": "^3.1.6",
```
Comment (maskmaster) [inline package.json:22]: Didn't we have a dependency on acorn to remove?

**Review by maskmaster — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (tscpp) [reply package.json:22]: No, we did actually not have acorn as a dependency, oddly enough

**Review by maskmaster — APPROVED** [review]

**Pull request merged**